adam-core/src/main/scala/org/bdgenomics/adam/rdd/variant/VariantContextRDD.scala

```diff
@@ -123,10 +124,14 @@ object VariantContextRDD extends Serializable {
 }
 
 case class DatasetBoundVariantContextRDD private[rdd] (
-    dataset: Dataset[VariantContextProduct],
-    sequences: SequenceDictionary,
-    @transient samples: Seq[Sample],
-    @transient headerLines: Seq[VCFHeaderLine] = DefaultHeaderLines.allHeaderLines) extends VariantContextRDD {
+  dataset: Dataset[VariantContextProduct],
+  sequences: SequenceDictionary,
+  @transient samples: Seq[Sample],
+  @transient headerLines: Seq[VCFHeaderLine] = DefaultHeaderLines.allHeaderLines,
+  override val isPartitioned: Boolean = true,
+  override val optPartitionBinSize: Option[Int] = Some(1000000),
+  override val optLookbackPartitions: Option[Int] = Some(1)) extends VariantContextRDD
+    with DatasetBoundGenomicDataset[VariantContext, VariantContextProduct, VariantContextRDD] {
```

Comment:
I'm not fully aware of the implications of adding this trait, thought it preferable to copying the `referenceRegionsToDatasetQueryString` and `filterByOverlappingRegions` methods

Reply:
This trait should be mixed in, so LGTM.